staff/tasks.py

```diff
@@ -1,10 +1,36 @@
 # Create your tasks here
 from __future__ import absolute_import, unicode_literals
-
 from celery import shared_task
+from home.models.models import DjangoUser, UserProfile, Group
+from home.models.salesforce import Contact
+from staff.staff_views_helper import create_django_user_from_contact
+
+
+@shared_task
+def add_salesforce_contacts_to_postgres():
+    """This method will identify contacts (students, teachers, and volunteers only)
+    who exist in the salesforce database but do not exist in the postgres database, and
+    add them to the local postgres database."""
+    contacts = Contact.objects.filter(title__in=["Student", "Teacher", "Volunteer"])
+    for contact in contacts:
+        if not DjangoUser.objects.filter(first_name=contact.first_name,
+                                         last_name=contact.last_name,
+                                         email=contact.email).exists():
+            create_django_user_from_contact(contact)
+
 
-"""
 @shared_task
-def hello():
-    print("It's a beautiful day in the neighborhood")
-"""
+def sync_userprofile_data_with_salesforce_data():
+    """This method will assure that contacts corresponding to existing django_users' userprofiles
+    will have matching salesforce_id/client_id fields in the postgres and salesforce databases, respectively"""
+    for user in UserProfile.objects.all():
+        django_user = DjangoUser.objects.get(id=user.user_id)
```

Comment:
It might be worth skipping rows where the salesforce id appears to be set correctly

Reply:
I think by "appears to be set correctly" you mean the entry in our db is not the default entry, and therefore was set correctly.  On the Salesforce end though, if someone changes a birthdate or name, then the client_id in salesforce (which is the salesforce_id in postgres) is auto-updated (tested this a bit ago), and this would create a mismatch.

Reply:
I wasn't aware that the salesforce id can change, this makes sense then.